extensions/wikia/PortableInfobox/services/Helpers/PortableInfoboxRenderServiceHelper.php

```diff
@@ -214,19 +215,51 @@ private function getThumbnail( $title ) {
 	}
 
 	/**
-	 * @desc get image size according to the width and height limitations:
+	 * @desc get image size to be passed to a thumbnailer.
+	 *
+	 * Return size according to the width and height limitations:
 	 * Height on desktop cannot be bigger than 500px
 	 * Width have to be adjusted to const for mobile or desktop infobox
+	 * if $wgPortableInfoboxCustomImageWidth is set, do not change max height
+	 * and set width to $wgPortableInfoboxCustomImageWidth.
 	 * @param $image
 	 * @return array width and height
 	 */
-	public function getAdjustedImageSize( $image ) {
+	public function getImageSizesForThumbnailer( $image ) {
+		global $wgPortableInfoboxCustomImageWidth;
+
 		if ( $this->isWikiaMobile() ) {
 			$width = self::MOBILE_THUMBNAIL_WIDTH;
 			$height = null;
-		} else {
+		} else if ( empty( $wgPortableInfoboxCustomImageWidth ) ) {
 			$height = min( self::MAX_DESKTOP_THUMBNAIL_HEIGHT, $image->getHeight() );
 			$width = self::DESKTOP_THUMBNAIL_WIDTH;
+		} else {
+			$height = $image->getHeight();
+			$width = $wgPortableInfoboxCustomImageWidth;
+		}
+
+		return [ 'height' => $height, 'width' => $width ];
+	}
+
+	/**
+	 * @desc if it's not a request from mobile skin and wgPortableInfoboxCustomImageWidth
+	 * is set, the $thumbnail->getWidth() can return some big value - we need
+	 * to adjust it to DESKTOP_THUMBNAIL_WIDTH to look good in the infobox.
+	 * Also, the $height have to be adjusted here to make image look good in infobox.
+	 * @param $thumbnail
+	 * @return array width and height which will be displayed i.e. in the width
+	 * and height properties of the img tag
+	 */
+	private function getImageSizesToDisplay( $thumbnail ) {
```

Comment:
This is a method that could IMO be public and tested. Any objections towards doing that?